src/toolkits/activity_classification/activity_classifier.cpp

```diff
@@ -837,10 +837,14 @@ void activity_classifier::init_train(
     for (size_t i = 1; i < gpu_names.size(); ++i) {
       gpu_names_string += ", " + gpu_names[i];
     }
-    logprogress_stream << "Using "
-                       << (gpu_names.size() > 1 ? "GPUs" : "GPU")
-                       << " to create model ("
-                       << gpu_names_string << ")";
+    if (gpu_names_string.find("AMD") != std::string::npos) {
```

Comment:
I don't think it's safe to conditionalize on "AMD". Why do we need to?

Reply:
Removed this and made it a condition on `\`